Sources/SpeziOnboarding/OnboardingIdentifiableViewModifier.swift

```diff
@@ -0,0 +1,39 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2024 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import SwiftUI
+
+/// Wrap `Content` `View` in an `Identifiable` `View`.
+public struct OnboardingIdentifiableView<Content>: View, Identifiable where Content: View {
+    /// Unique identifier of the wrapped `View`.
+    public let id: String
+    /// Wrapped `View`.
+    public var body: Content
+}
+
+struct OnboardingIdentifiableViewModifier: ViewModifier {
+    let identifier: String
+
+    func body(content: Content) -> some View {
+        OnboardingIdentifiableView(
+            id: self.identifier,
+            body: content
+        )
+    }
+}
+
+
+extension View {
+    /// `ViewModifier` assigning an identifier to the `View` it is applied to.
+    /// When applying this modifier repeatedly, the outermost ``id(_:)`` counts.
+    /// - Parameters:
+    ///   - identifier: The `String` identifier given to the view.
+    public func id(_ identifier: String) -> some View {
+        modifier(OnboardingIdentifiableViewModifier(identifier: identifier))
```

Comment:
To my understanding this modifier is currently not tested and I have the suspicion that this doesn't work as this has a return type of type `ModifiedContent<Self, OnboardingIdentifiableViewModifier> which does conform to View, but doesn't conform to Identifiable. We might need to add a conditional conformance to ModifiedContent if `Content` is Identifiable, so we can detect a modified, identifiable view (e.g., also if there are other modifiers used with a normal view etc).

Also, do we ever consider Identifiable views in the `@OnboardingViewBuilder`, doesn't currently seem like it?

Reply:
Thanks for pointing that out, it was naive of me to not have a test for this!
The reason why we cannot use `Content: Identifiable` as a type constraint here is that `Content` refers to the `View` **before** the modifier is applied, thus the `View` that we want to make `Identifiable`.
Therefore I extended `ModifiedContent` to be `Identifiable` when its `Modifier` is `Identifiable` (see code).

Reply:
For what exact reason do you want to consider `Identifiable` views in the `OnboardingViewBuilder`?

Reply:
> Therefore I extended ModifiedContent to be Identifiable when its Modifier is Identifiable (see code).

Sounds great. Yes, I confused the `Content` associated type. Making the modifier itself conform to Identifiable is the way to go. Thank you.

> For what exact reason do you want to consider Identifiable views in the OnboardingViewBuilder?

My approach would have been to construct `OnboardingStepIdentifier`s already within the `OnboardingViewBuilder`. This would eliminate the dynamic type check within the `OnboardingStepIdentifier`, though would also require to create a dedicated component type that stores this within the collection, instead of just using `[any View]`. 

I'm not sure if it is just me learning Swift at a time where dynamic type checks using protocols that contained associated type requirements weren't possible 🙊.

Just to illustrate my approach, you would have added expression builders as below. No need to implement, just to illustrate my train of thought here.
```swift
public static func buildExpression<Content: View & Identifiable>(_ expression: Content) -> [OnboardingView]

// bonus, would eliminate the Identifiable extension on ModifiedContent.
public static func buildExpression<Content: View, Modifier: Identifiable>(_ expression: ModifiedContent<Content, Modifier>) -> [OnboardingView]

public static func buildExpression<Content: View>(_ expression: Content) -> [OnboardingView]
```